include/cantera/kinetics/RxnRates.h

```diff
@@ -84,6 +84,110 @@ class Arrhenius
     doublereal m_logA, m_b, m_E, m_A;
 };
 
+//! Blowers Masel reaction rate type depends on enthalpy 
+/**
+ * The Blowers Masel approximation is written by [Paul Blowrs,
+ * Rich Masel(2004). Engineering Approximations For Activation Energies in Hydrogen
+ * Transfer Reactions. Eq (10)-(11)] to adjust the activation energy based on
+ * enthalpy change of a reaction:
+ *
+ *   \f[
+ *        Ea = 0  if DeltaH < -4E_0
+ *        Ea = DeltaH   if DeltaH > 4E_0
+ *        Ea = \frac{(w_0 + DeltaH / 2)(V_P - 2w_0 + DeltaH)^2}{(V_P^^2 - 4w_0^2 + DeltaH^2)}
+ *   \f] 
+ * where
+ *   \f[
+ *        V_P = \frac{2w_0 (w_0 + E_0)}{w_0 - E_0}
+ *   \f]
+ * and \f$ w_0 \f$ is  theaverage of the bond energy of the 
+ * bond breaking and that being formed, which can be approximated as
+ * arbitrary high value like 1000kJ/mol as long as \f$ w_0 >= 2 E_0 \f$
+ * The rate constant then can be calculated by Arrhenius function.
+ *   \f[
+ *        k_f =  A T^b \exp (-Ea/RT)
+ *   \f]
+ */
+
+class BlowersMasel
+{
+public:
+    //! Default constructor.
+    BlowersMasel();
+
+    /// Constructor.
+    /// @param A pre-exponential. The unit system is
+    ///     (kmol, m, s). The actual units depend on the reaction
+    ///     order and the dimensionality (surface or bulk).
+    /// @param b Temperature exponent. Non-dimensional.
+    /// @param E0 Activation energy in temperature units. Kelvin.
+    /// @param w bond energy of the bond being formed or broken, in temperature unts. Kelvin.
+
+    BlowersMasel(doublereal A, doublereal b, doublereal E0, doublereal w);
+
+    //! Update concentration-dependent parts of the rate coefficient.
+    /*!
+     *   For this class, there are no concentration-dependent parts, so this
+     *   method does nothing.
+     */
+    void update_C(const doublereal* c) {
+    }
+
+    /**
+     * Update the value the rate constant.
+     *
+     * This function returns the actual value of the rate constant. It can be
+     * safely called for negative values of the pre-exponential factor.
+     */
+    doublereal updateRC(doublereal logT, doublereal recipT, doublereal deltaH) {
+        m_E = activationEnergy_R(deltaH);
+        return m_A * std::exp(m_b * logT - m_E * recipT);
```

Comment:
It looks like the member variable `m_E` is only used here, so I think it could be eliminated. (in which case both this and `activationEnergy_R` could be declared `const`).